sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -53,12 +53,12 @@ case class SortMergeJoinExec(
     // For inner join, orders of both sides keys should be kept.
     case _: InnerLike =>
       val leftKeyOrdering = getKeyOrdering(leftKeys, left.outputOrdering)
-      val rightKeyOrdering = getKeyOrdering(rightKeys, right.outputOrdering)
+      val rightKeyOrdering = getKeyOrdering(rightKeys, right.outputOrdering, false)
       leftKeyOrdering.zip(rightKeyOrdering).map { case (lKey, rKey) =>
         // Also add expressions from right side sort order
         val sameOrderExpressions = ExpressionSet(lKey.sameOrderExpressions ++ rKey.children)
         SortOrder(lKey.child, Ascending, sameOrderExpressions.toSeq)
-      }
+      } ++ leftKeyOrdering.drop(rightKeyOrdering.size)
```

Comment:
how about the extra right side key ordering?

Reply:
For the buffered side may spill to disk when there are too many matching rows in the scanner.
I'm not sure if the rows reading from the spill file have the same ordering with the input rows.

Reply:
Added comments